client/src/components/StreamField/blocks/StreamBlock.js

```diff
@@ -98,12 +98,40 @@ class StreamBlockMenu {
         });
       });
     });
+
+    // Disable buttons for any disabled block types
+    this.disabledBlockTypes.forEach(blockType => {
+      $(`button.action-add-block-${h(blockType)}`, this.innerContainer).attr('disabled', 'true');
+    });
   }
 
   setIndex(newIndex) {
     this.index = newIndex;
   }
 
+  setNewBlockRestrictions(canAddBlock, disabledBlockTypes) {
+    this.canAddBlock = canAddBlock;
+    this.disabledBlockTypes = disabledBlockTypes;
```

Comment:
We should set initial values for these in the constructor so that we're not implicitly relying on the parent block to call this before renderMenu or open. ~~(Additionally we might want to split this into separate setter functions for canAddBlock and disabledBlockTypes, since canAddBlock belongs on the base InsertionControl class whereas disabledBlockTypes is specific to the stream menu)~~ that can come later - for now we can keep this StreamBlock-specific